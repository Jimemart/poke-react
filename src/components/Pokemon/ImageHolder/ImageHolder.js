import React from "react"
import PropTypes from "prop-types"
import nanoid from "nanoid"
import { FlexContainer } from "../../../UI"

export const ImageHolder = props => {
  const { imgs } = props
  const border = "2px solid #ce0808"
  return (
    <FlexContainer width="40%" border={border}>
      {imgs.map((elem, i) => (
        <div
          key={`img-${nanoid()}`}
          style={{ borderRight: !i ? border : "none" }}
                >
          <img src={elem} alt={`img-${i}`} />
        </div>
      ))}
    </FlexContainer>
  )
}

ImageHolder.propTypes = {
  imgs: PropTypes.arrayOf(PropTypes.string).isRequired,
}
