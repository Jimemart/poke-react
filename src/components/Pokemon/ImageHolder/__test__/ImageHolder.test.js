import React from "react";
import renderer from "react-test-renderer";
import { ImageHolder } from "../ImageHolder";

describe("ImageHolder Component", () => {
  const props = {
    imgs: ["foo", "bar"]
  };

  it("should render the two images", () => {
    const tree = renderer.create(
      <ImageHolder {...props}>
        <div>Some nested content</div>
      </ImageHolder>
    );

    expect(tree).toMatchSnapshot();
  });
});
