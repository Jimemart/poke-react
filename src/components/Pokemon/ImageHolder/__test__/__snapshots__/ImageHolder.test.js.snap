// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot_1 1`] = `
<div
  className="sc-bdVaJa jyDCco"
  width="40%"
>
  <div
    style={
      Object {
        "borderRight": "2px solid #ce0808",
      }
    }
  >
    <img
      alt="img-0"
      src="foo"
    />
  </div>
  <div
    style={
      Object {
        "borderRight": "none",
      }
    }
  >
    <img
      alt="img-1"
      src="bar"
    />
  </div>
</div>
`;
